logstash-core/spec/logstash/agent/metrics_spec.rb

```diff
@@ -73,6 +73,15 @@ def mhash(*path_elements)
       expect(mval(:stats, :reloads, :successes)).to eq(0)
       expect(mval(:stats, :reloads, :failures)).to eq(0)
     end
+
+    it "should validate empty flow metrics" do
+      expect(mval(:stats, :flow, :input_throughput)).to eq({})
+      expect(mval(:stats, :flow, :output_throughput)).to eq({})
+      expect(mval(:stats, :flow, :filter_throughput)).to eq({})
+      expect(mval(:stats, :flow, :queue_backpressure)).to eq({})
+      expect(mval(:stats, :flow, :worker_concurrency)).to eq({})
+    end
```

Comment:
2 things:
 - imperative language. we specify what it _does_, not what it _should_ do
 - future-proof by specifying that it should be a Hash, without specifying that it should be an empty Hash, so that if/when we do capture them we don't break our specification.

```suggestion
    it "makes the top-level flow metrics available available" do
      expect(mval(:stats, :flow, :input_throughput)).to be_a_kind_of Hash
      expect(mval(:stats, :flow, :output_throughput)).to be_a_kind_of Hash
      expect(mval(:stats, :flow, :filter_throughput)).to be_a_kind_of Hash
      expect(mval(:stats, :flow, :queue_backpressure)).to be_a_kind_of Hash
      expect(mval(:stats, :flow, :worker_concurrency)).to be_a_kind_of Hash
    end
```